recipes/lzma_sdk/9.20/conanfile.py

```diff
@@ -14,86 +16,99 @@
+
     def build_requirements(self):
-        if self.settings.compiler != "Visual Studio" and tools.os_info.is_windows and os.environ.get("CONAN_BASH_PATH", None) is None:
-            self.build_requires("msys2/20190524")
+        if self.settings.compiler != "Visual Studio" and self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+
+    def package_id(self):
+        del self.info.settings.compiler
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.unlink(os.path.join(self.source_folder, "7zr.exe"))
-        os.unlink(os.path.join(self.source_folder, "lzma.exe"))
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder)
+        os.unlink(os.path.join(self._source_subfolder, "7zr.exe"))
+        os.unlink(os.path.join(self._source_subfolder, "lzma.exe"))
 
     @property
     def _msvc_build_dirs(self):
         return (
-            [os.path.join("C","Util","7z"), [["7zDec.exe"], []]],
-            [os.path.join("C","Util","Lzma"), [["LZMAc.exe"], []]],
-            [os.path.join("CPP","7zip","UI","Console"), [["7z.exe"], []]],
-            [os.path.join("CPP","7zip","Bundles","Alone7z"), [["7zr.exe"], []]],
-            [os.path.join("CPP","7zip","Bundles","LzmaCon"), [["lzma.exe"], []]],
+            (os.path.join(self._source_subfolder, "C", "Util", "7z"), "7zDec.exe"),
+            (os.path.join(self._source_subfolder, "C", "Util", "Lzma"), "LZMAc.exe"),
+            (os.path.join(self._source_subfolder, "CPP", "7zip", "UI", "Console"), "7z.exe"),
+            (os.path.join(self._source_subfolder, "CPP", "7zip","Bundles", "Alone7z"), "7zr.exe"),
+            (os.path.join(self._source_subfolder, "CPP", "7zip", "Bundles", "LzmaCon"), "lzma.exe"),
         )
 
     @property
     def _msvc_cpu(self):
         return {
-            'x86_64': 'AMD64',
-            'x86': 'x86',
+            "x86_64": "AMD64",
+            "x86": "x86",
         }[str(self.settings.arch)]
 
     @property
     def _autotools_build_dirs(self):
         es = ".exe" if self.settings.os == "Windows" else ""
         return (
-            [os.path.join("C","Util","7z"), [["7zDec{}".format(es)], []]],
-            [os.path.join("CPP","7zip","Bundles","LzmaCon"), [["lzma{}".format(es)],[]]]
+            (os.path.join(self._source_subfolder, "C", "Util", "7z"), "7zDec{}".format(es)),
+            (os.path.join(self._source_subfolder, "CPP", "7zip", "Bundles", "LzmaCon"), "lzma{}".format(es)),
         )
 
     def _build_msvc(self):
         for make_dir, _ in self._msvc_build_dirs:
-            with tools.vcvars(self.settings):
+            with tools.vcvars(self):
                 with tools.chdir(make_dir):
-                    self.run("nmake /f makefile NEW_COMPILER=1 CPU=%s" % self._msvc_cpu)
+                    self.run("nmake /f makefile NEW_COMPILER=1 CPU={}".format(self._msvc_cpu))
```

Comment:
VisualStudioBuildEnvironment(self).vars should be added to environment, otherwise profile is not honored

Reply:
Like 911d4fc42 ?

Reply:
yes

Reply:
Is there anything we can do (CI side, hooks) to ensure it? I don't like to see a successful build if something as important as the profile is not honored 🤔

Reply:
This is a binary only package.
You can try to analyze the binaries.

Reply:
This is something we were thinking for cross-building scenarios in the test_package. It would work for native ones, as well. I suppose running tools like `otool` or similar doesn't add overhead to the build and can be very convenient. It could even be a _hook_ running after `package` on all the binaries in the package folder 🤔